BUG - Memory leaks in input/readline

### Bug description

The readline function run with errors and memory leaks.

### Steps to reproduce

1. Clone in commit 1647f21
2. Add some use of inputs in src-server/app/main.c
3. Run with:

```bash
rm -rf "/tmp/scorpion-tests/"scorpionc2 "/tmp/scorpion-tests/"
mkdir -p "/tmp/scorpion-tests/"
gcc -Isrc-server/domain/encoders/services/xor -Isrc-server/domain/encoders/types -Isrc-server/shared/types -Isrc-server/app/cli/colors -Isrc-server/app/cli/loading -Isrc-server/app/cli/logs -Isrc-server/app/cli/ui/box -Isrc-server/app/cli/input/readMode -Isrc-server/app/cli/input/safeget -Isrc-server/infra/fs -Isrc-server/app/cli/input  -Og -dA -dD -ggdb -Wall -Wextra -Wformat=2 -Wshadow -Wundef -o "/tmp/scorpion-tests/"scorpionc2 src-server/app/main.c src-server/app/cli/logs/main.c src-server/shared/utils/random/main.c src-server/shared/utils/hash/main.c src-server/domain/encoders/services/main.c src-server/domain/encoders/services/xor/main.c src-server/app/cli/ui/box/main.c src-server/app/cli/input/readMode/main.c src-server/app/cli/input/safeget/main.c src-server/app/cli/input/main.c src-server/infra/fs/main.c
"/usr/bin/valgrind" --leak-check=summary --errors-for-leak-kinds=definite -s "/tmp/scorpion-tests/"scorpionc2

```

### Expected behavior

Run normally and handle correctly the user input

### Actual behavior

Run with errors:

```
==250288== 1 errors in context 1 of 1:
==250288== Invalid read of size 1
==250288==    at 0x484ED16: strlen (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==250288==    by 0x48ECD30: __vfprintf_internal (vfprintf-internal.c:1517)
==250288==    by 0x48F7988: __vsprintf_internal (iovsprintf.c:95)
==250288==    by 0x10ACED: vsprintf (stdio2.h:52)
==250288==    by 0x10ACED: infof (main.c:91)
==250288==    by 0x10A966: main (main.c:121)
==250288==  Address 0x4ab5f90 is 0 bytes after a block of size 0 alloc'd
==250288==    at 0x4848899: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==250288==    by 0x10CF4E: readline (main.c:33)
==250288==    by 0x10A946: main (main.c:115)
==250288== 
==250288== ERROR SUMMARY: 1 errors from 1 contexts (suppressed: 0 from 0)
```

### Logs / stack traces

```shell
➜  ScorpionC2 git:(main) ✗ make test-valgrind
rm -rf "/tmp/scorpion-tests/"scorpionc2 "/tmp/scorpion-tests/"
mkdir -p "/tmp/scorpion-tests/"
gcc -Isrc-server/domain/encoders/services/xor -Isrc-server/domain/encoders/types -Isrc-server/shared/types -Isrc-server/app/cli/colors -Isrc-server/app/cli/loading -Isrc-server/app/cli/logs -Isrc-server/app/cli/ui/box -Isrc-server/app/cli/input/readMode -Isrc-server/app/cli/input/safeget -Isrc-server/infra/fs  -Og -dA -dD -ggdb -o "/tmp/scorpion-tests/"scorpionc2 src-server/app/main.c src-server/app/cli/logs/main.c src-server/shared/utils/random/main.c src-server/shared/utils/hash/main.c src-server/domain/encoders/services/main.c src-server/domain/encoders/services/xor/main.c src-server/app/cli/ui/box/main.c src-server/app/cli/input/readMode/main.c src-server/app/cli/input/safeget/main.c src-server/infra/fs/main.c 
/usr/bin/ld: /tmp/ccUvooP8.o: aviso: relocalização contra "Input" em secção só de leitura ".text"
/usr/bin/ld: /tmp/ccUvooP8.o: na função "main":
/tmp/ScorpionC2/src-server/app/main.c:115: referência não definida para "Input"
/usr/bin/ld: aviso: a criar DT_TEXTREL num PIE.
collect2: error: ld returned 1 exit status
make: *** [Makefile:93: test-valgrind] Erro 1
➜  ScorpionC2 git:(main) ✗ cat src-server/app/cli/input/main.c
//
// Copyright (c) 2026-Present ScorpionC2 public-person "Lucas de Moraes Claro" and all anonymous contributors. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for details.
//

#include "main.h"
#include "safeget/main.h"
#include "../../../infra/fs/main.h"
#include "../logs/main.h"
#include <string.h>
#include <stdlib.h>

void readline(InputSettings conf, string_t *out) {
    string_t prompt = conf.prompt;
    string_t histPath = conf.histPath;
    int histLimit = conf.histLimit;
    
    printf("%s", prompt.s);
    fflush(stdout);
    string_t input = SafeGet.ucread(conf.histPath, prompt.len - 1);
    bytes_t inputRaw = {
        .len = input.len,
        .b = (uchar_t*)input.s
        
    };
    
    if (Files.appendFile(histPath, &inputRaw) != 0) {
        Logger.warnln("Can't write last user input to history file");
        
    }
    
    out->len = input.len;
    out->s = malloc(out->len);
    memcpy(out->s, input.s, out->len);
    
    free(input.s);
    return;
    
};

InputInstance Input = {
    .readline = readline
    
};%                                                                                                                                                                                         ➜  ScorpionC2 git:(main) ✗ rm -rf "/tmp/scorpion-tests/"scorpionc2 "/tmp/scorpion-tests/"
mkdir -p "/tmp/scorpion-tests/"
gcc -Isrc-server/domain/encoders/services/xor -Isrc-server/domain/encoders/types -Isrc-server/shared/types -Isrc-server/app/cli/colors -Isrc-server/app/cli/loading -Isrc-server/app/cli/logs -Isrc-server/app/cli/ui/box -Isrc-server/app/cli/input/readMode -Isrc-server/app/cli/input/safeget -Isrc-server/infra/fs -Isrc-server/app/cli/input  -Og -dA -dD -ggdb -Wall -Wextra -Wformat=2 -Wshadow -Wundef -o "/tmp/scorpion-tests/"scorpionc2 src-server/app/main.c src-server/app/cli/logs/main.c src-server/shared/utils/random/main.c src-server/shared/utils/hash/main.c src-server/domain/encoders/services/main.c src-server/domain/encoders/services/xor/main.c src-server/app/cli/ui/box/main.c src-server/app/cli/input/readMode/main.c src-server/app/cli/input/safeget/main.c src-server/app/cli/input/main.c src-server/infra/fs/main.c
"/usr/bin/valgrind" --leak-check=summary --errors-for-leak-kinds=definite -s "/tmp/scorpion-tests/"scorpionc2

src-server/app/cli/logs/main.c: In function ‘bytesf’:
src-server/app/cli/logs/main.c:169:35: warning: comparison of integer expressions of different signedness: ‘int’ and ‘size_t’ {aka ‘long unsigned int’} [-Wsign-compare]
  169 |                 for (int i = 1; i < bytes->len; i++) {
      |                                   ^
src-server/app/cli/logs/main.c:183:31: warning: comparison of integer expressions of different signedness: ‘int’ and ‘size_t’ {aka ‘long unsigned int’} [-Wsign-compare]
  183 |             for (int i = 1; i < bytes->len; i++) {
      |                               ^
src-server/shared/utils/random/main.c: In function ‘Randomrseed’:
src-server/shared/utils/random/main.c:35:98: warning: suggest parentheses around arithmetic in operand of ‘^’ [-Wparentheses]
   35 | time(NULL) ^ (uint32_t)getpid())) ^ (uint32_t)(seed / (uint32_t)getpid()) + (time(NULL) - (uint32_t)getpid())) * 2677735761u)) % (((((seed * seed) ^ seed) - 25) + 569821785) >> (seed % 8));
      |                                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

src-server/shared/utils/random/main.c: In function ‘Randomrand’:
src-server/shared/utils/random/main.c:43:53: warning: suggest parentheses around arithmetic in operand of ‘^’ [-Wparentheses]
   43 |         s = Randomrseed(seed_g ^ (3290751 * seed_g) + (Randomrseed(seed_g * 49091339) >> 15));
      |                                  ~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src-server/shared/utils/hash/main.c: In function ‘HashScorpion2048x’:
src-server/shared/utils/hash/main.c:41:27: warning: comparison of integer expressions of different signedness: ‘int’ and ‘size_t’ {aka ‘long unsigned int’} [-Wsign-compare]
   41 |     for (int i = 0; i + 3 < out.len; i++) {
      |                           ^
src-server/shared/utils/hash/main.c: In function ‘HashScorpion1024x’:
src-server/shared/utils/hash/main.c:94:27: warning: comparison of integer expressions of different signedness: ‘int’ and ‘size_t’ {aka ‘long unsigned int’} [-Wsign-compare]
   94 |     for (int i = 0; i + 3 < out.len; i++) {
      |                           ^
src-server/shared/utils/hash/main.c: In function ‘HashScorpion512x’:
src-server/shared/utils/hash/main.c:147:27: warning: comparison of integer expressions of different signedness: ‘int’ and ‘size_t’ {aka ‘long unsigned int’} [-Wsign-compare]
  147 |     for (int i = 0; i + 3 < out.len; i++) {
      |                           ^
src-server/shared/utils/hash/main.c: In function ‘HashScorpion256x’:
src-server/shared/utils/hash/main.c:200:27: warning: comparison of integer expressions of different signedness: ‘int’ and ‘size_t’ {aka ‘long unsigned int’} [-Wsign-compare]
  200 |     for (int i = 0; i + 3 < out.len; i++) {
      |                           ^
src-server/shared/utils/hash/main.c: In function ‘HashScorpion128x’:
src-server/shared/utils/hash/main.c:253:27: warning: comparison of integer expressions of different signedness: ‘int’ and ‘size_t’ {aka ‘long unsigned int’} [-Wsign-compare]
  253 |     for (int i = 0; i + 3 < out.len; i++) {
      |                           ^
src-server/domain/encoders/services/xor/main.c: In function ‘xor’:
src-server/domain/encoders/services/xor/main.c:42:23: warning: comparison of integer expressions of different signedness: ‘int’ and ‘size_t’ {aka ‘long unsigned int’} [-Wsign-compare]
   42 |     for (int i = 0; i < src.len; i++) {
      |                       ^
src-server/app/cli/input/safeget/main.c: In function ‘ucgetchar’:
src-server/app/cli/input/safeget/main.c:28:22: warning: unused parameter ‘stream’ [-Wunused-parameter]
   28 | char ucgetchar(FILE *stream) {
      |                ~~~~~~^~~~~~
src-server/app/cli/input/safeget/main.c: In function ‘ucread’:
src-server/app/cli/input/safeget/main.c:169:9: warning: ‘buf’ is used uninitialized [-Wuninitialized]
  169 |     if (ucfgets(buf, histPath, promptSize) == NULL) {
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src-server/app/cli/input/safeget/main.c: In function ‘ucfgets’:
src-server/app/cli/input/safeget/main.c:152:11: warning: ‘_n’ may be used uninitialized in this function [-Wmaybe-uninitialized]
  152 |         _n++;
      |         ~~^~
src-server/app/cli/input/main.c: In function ‘readline’:
src-server/app/cli/input/main.c:16:9: warning: unused variable ‘histLimit’ [-Wunused-variable]
   16 |     int histLimit = conf.histLimit;
      |         ^~~~~~~~~
==250288== Memcheck, a memory error detector
==250288== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==250288== Using Valgrind-3.18.1 and LibVEX; rerun with -h for copyright info
==250288== Command: /tmp/scorpion-tests/scorpionc2
==250288== 
Bytes: [68, 6f, 77, 6f, 0]
Bytes with 0x delimiter: [0x68, 0x6f, 0x77, 0x6f, 0]
Address to string: 0x1ffefff9a3
[DEBUG] Debugging: howo
[!] Warning: howo
[?] Information: howo
[*] Error: howo
[?] Random number: 3885748427
[?] Random number (0-255): 213
Xored string "howo": [0xfb, 0x1, 0x6c, 0xdf, 0xc7, 0xef, 0x2b, 0x1a, 0x50, 0x6b, 0x16, 0xde, 0xa, 0x5c, 0x46, 0xea, 0x6f, 0xcb, 0xd3, 0x37, 0xff, 0xb6, 0xaf, 0xcc, 0x13, 0x28, 0xa6, 0x8e, 0xb4, 0x3d, 0x64, 0xf0, 0xef, 0x3f, 0x53, 0xa9, 0x9c, 0xa9, 0x35, 0x22, 0x13, 0x40, 0xd1, 0xde, 0xe1, 0x19, 0x8b, 0xd4, 0x8b, 0xff, 0x52, 0x15, 0xa0, 0xf3, 0x37, 0xcf, 0x70, 0x34, 0xa4, 0xd9, 0x8b, 0xdf, 0x94, 0xb5, 0xca, 0xb7, 0x85, 0x50, 0xd3, 0x38, 0xfc, 0x34, 0xc9, 0x79, 0xbe, 0x49, 0x12, 0x82, 0x95, 0xad, 0xa9, 0x68, 0x2c, 0xdd, 0xc0, 0x92, 0x18, 0x37, 0xca, 0x1, 0xaa, 0x81, 0xac, 0x63, 0x1e, 0x26, 0xe7, 0x93, 0x2d, 0x21, 0xdf, 0x8f, 0xd5, 0xd8, 0xfc, 0x38, 0x90, 0xa0, 0x23, 0xd6, 0xa3, 0x2a, 0x3f, 0x2, 0x27, 0xff, 0x40, 0x36, 0xec, 0xc2, 0xd, 0x7a, 0x67, 0x8b, 0x9e, 0x52, 0x1d, 0x1a, 0xb9, 0x4e, 0xa6, 0x4e, 0xd1]
Unxored string "howo": [0x68, 0x6f, 0x77, 0x6f, 0]
Xored string with nonce size 16 "howo": [0x4c, 0x4e, 0x64, 0x7a, 0xb9, 0x9b, 0xea, 0x1a, 0x79, 0x27, 0xc3, 0x1b, 0xe8, 0x70, 0x35, 0x23, 0x96, 0x2f, 0x89, 0x2f, 0xfe]
Unxored string with nonce size 16 "howo": [0x68, 0x6f, 0x77, 0x6f, 0]
Xored string with hash algorithm scorpion512x "howo": [0x15, 0x55, 0x4b, 0xcd, 0x6b, 0xd3, 0x10, 0xb6, 0xda, 0xd5, 0x4, 0x6c, 0xbb, 0xea, 0xf4, 0x8c, 0xa3, 0xf8, 0xa6, 0x9c, 0x46, 0x39, 0x14, 0xd6, 0x86, 0x4c, 0x32, 0x5d, 0x57, 0x9b, 0x2b, 0x5c, 0x58, 0x6, 0xe1, 0x42, 0x1e, 0xf4, 0x1, 0x7, 0x11, 0xeb, 0x59, 0x96, 0x22, 0xaf, 0x67, 0x2c, 0x16, 0x9e, 0xa5, 0xb6, 0x9b, 0x60, 0xfc, 0x8f, 0xd7, 0x94, 0x3d, 0x8e, 0, 0x53, 0xe1, 0x7e, 0xcf, 0xd6, 0x84, 0x95, 0x5b, 0xad, 0x75, 0xa4, 0x81, 0xd6, 0x26, 0xe9, 0xa0, 0x3a, 0x7c, 0xc0, 0x8c, 0x9, 0xe7, 0x48, 0xa3, 0x59, 0x77, 0x4, 0x66, 0x3e, 0xd9, 0x27, 0xe9, 0x40, 0x73, 0x37, 0x96, 0x58, 0x7a, 0xf1, 0x6a, 0x7b, 0xf5, 0xd9, 0xcb, 0x2, 0x95, 0x79, 0x3d, 0xb, 0x93, 0x91, 0x87, 0x1a, 0x31, 0xd3, 0x22, 0x7d, 0x23, 0xee, 0x30, 0xcf, 0x80, 0x10, 0x66, 0x69, 0xb1, 0xdb, 0x39, 0x88, 0x26, 0x88, 0x51]
Unored string with hash algorithm scorpion512x "howo": [0x68, 0x6f, 0x77, 0x6f, 0]


Hash for string "howo" (djb2): [0x22, 0xdb, 0x97, 0xf, 0x31, 0, 0, 0]

Hash for string "howo" (scorpion128x): [0x15, 0x1b, 0x1c, 0x1b, 0x10, 0x15, 0x1b, 0x1c, 0x1b, 0x10, 0x15, 0x1b, 0x1c, 0x1b, 0x10, 0x15, 0x1b, 0x1c, 0x1b, 0x10, 0x15, 0x1b, 0x1c, 0x1b, 0x10, 0x15, 0x1b, 0x1c, 0x1b, 0x10, 0x15, 0x1b, 0x1c, 0x1b, 0x10, 0x15, 0x1b, 0x1c, 0x1b, 0x10, 0x15, 0x1b, 0x1c, 0x1b, 0x10, 0x15, 0x1b, 0x1c, 0x1b, 0x10, 0x15, 0x1b, 0x1c, 0x1b, 0x10, 0x15, 0x1b, 0x1c, 0x1b, 0x10, 0x15, 0x1b, 0x1c, 0x1b, 0x10, 0x15, 0x1b, 0x1c, 0x1b, 0x10, 0x15, 0x1b, 0x1c, 0x1b, 0x10, 0x15, 0x1b, 0x1c, 0x1b, 0x10, 0x15, 0x1b, 0x1c, 0x1b, 0x10, 0x15, 0x1b, 0x1c, 0x1b, 0x10, 0x15, 0x1b, 0x1c, 0x1b, 0x10, 0x15, 0x1b, 0x1c, 0x1b, 0x10, 0x15, 0x1b, 0x1c, 0x1b, 0x10, 0x15, 0x1b, 0x1c, 0x1b, 0x10, 0x15, 0x1b, 0x1c, 0x1b, 0x10, 0x15, 0x1b, 0x1c, 0x1b, 0x10, 0x15, 0x1b, 0x1c, 0x1b, 0x10, 0xa0, 0xb4, 0x5e]

Hash for string "howo" (scorpion256x): [0xb7, 0xa7, 0xf7, 0xa7, 0xc6, 0xb7, 0xa7, 0xf7, 0xa7, 0xc6, 0xb7, 0xa7, 0xf7, 0xa7, 0xc6, 0xb7, 0xa7, 0xf7, 0xa7, 0xc6, 0xb7, 0xa7, 0xf7, 0xa7, 0xc6, 0xb7, 0xa7, 0xf7, 0xa7, 0xc6, 0xb7, 0xa7, 0xf7, 0xa7, 0xc6, 0xb7, 0xa7, 0xf7, 0xa7, 0xc6, 0xb7, 0xa7, 0xf7, 0xa7, 0xc6, 0xb7, 0xa7, 0xf7, 0xa7, 0xc6, 0xb7, 0xa7, 0xf7, 0xa7, 0xc6, 0xb7, 0xa7, 0xf7, 0xa7, 0xc6, 0xb7, 0xa7, 0xf7, 0xa7, 0xc6, 0xb7, 0xa7, 0xf7, 0xa7, 0xc6, 0xb7, 0xa7, 0xf7, 0xa7, 0xc6, 0xb7, 0xa7, 0xf7, 0xa7, 0xc6, 0xb7, 0xa7, 0xf7, 0xa7, 0xc6, 0xb7, 0xa7, 0xf7, 0xa7, 0xc6, 0xb7, 0xa7, 0xf7, 0xa7, 0xc6, 0xb7, 0xa7, 0xf7, 0xa7, 0xc6, 0xb7, 0xa7, 0xf7, 0xa7, 0xc6, 0xb7, 0xa7, 0xf7, 0xa7, 0xc6, 0xb7, 0xa7, 0xf7, 0xa7, 0xc6, 0xb7, 0xa7, 0xf7, 0xa7, 0xc6, 0xb7, 0xa7, 0xf7, 0xa7, 0xc6, 0xb7, 0xa7, 0xf7, 0xa7, 0xc6, 0xb7, 0xa7, 0xf7, 0xa7, 0xc6, 0xb7, 0xa7, 0xf7, 0xa7, 0xc6, 0xb7, 0xa7, 0xf7, 0xa7, 0xc6, 0xb7, 0xa7, 0xf7, 0xa7, 0xc6, 0xb7, 0xa7, 0xf7, 0xa7, 0xc6, 0xb7, 0xa7, 0xf7, 0xa7, 0xc6, 0xb7, 0xa7, 0xf7, 0xa7, 0xc6, 0xb7, 0xa7, 0xf7, 0xa7, 0xc6, 0xb7, 0xa7, 0xf7, 0xa7, 0xc6, 0xb7, 0xa7, 0xf7, 0xa7, 0xc6, 0xb7, 0xa7, 0xf7, 0xa7, 0xc6, 0xb7, 0xa7, 0xf7, 0xa7, 0xc6, 0xb7, 0xa7, 0xf7, 0xa7, 0xc6, 0xb7, 0xa7, 0xf7, 0xa7, 0xc6, 0xb7, 0xa7, 0xf7, 0xa7, 0xc6, 0xb7, 0xa7, 0xf7, 0xa7, 0xc6, 0xb7, 0xa7, 0xf7, 0xa7, 0xc6, 0xb7, 0xa7, 0xf7, 0xa7, 0xc6, 0xb7, 0xa7, 0xf7, 0xa7, 0xc6, 0xb7, 0xa7, 0xf7, 0xa7, 0xc6, 0xb7, 0xa7, 0xf7, 0xa7, 0xc6, 0xb7, 0xa7, 0xf7, 0xa7, 0xc6, 0xb7, 0xa7, 0xf7, 0xa7, 0xc6, 0xb7, 0xa7, 0xf7, 0xa7, 0xc6, 0xb7, 0xa7, 0xf7, 0x8c, 0xfa, 0x36]

Hash for string "howo" (scorpion512x): [0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0x27, 0xd5, 0xce, 0x50, 0xce, 0xad, 0x42, 0x40]

Hash for string "howo" (scorpion1024x): [0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xc8, 0x68, 0xc8, 0xd7, 0x32, 0xbd, 0x9b, 0x7c]

Hash for string "howo" (scorpion2048x): [0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x78, 0x8a, 0x78, 0x8a, 0xe9, 0x7, 0x72, 0x83]
[?] Gradient Box:
╭──────────────────────╮
│                      │
│                      │
│                      │
│                      │
│                      │
│                      │
│                      │
│                      │
│                      │
│                      │
╰──────────────────────╯
c2 -> ==250288== Invalid read of size 1
==250288==    at 0x484ED16: strlen (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==250288==    by 0x48ECD30: __vfprintf_internal (vfprintf-internal.c:1517)
==250288==    by 0x48F7988: __vsprintf_internal (iovsprintf.c:95)
==250288==    by 0x10ACED: vsprintf (stdio2.h:52)
==250288==    by 0x10ACED: infof (main.c:91)
==250288==    by 0x10A966: main (main.c:121)
==250288==  Address 0x4ab5f90 is 0 bytes after a block of size 0 alloc'd
==250288==    at 0x4848899: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==250288==    by 0x10CF4E: readline (main.c:33)
==250288==    by 0x10A946: main (main.c:115)
==250288== 
[?] Input: 
==250288== 
==250288== HEAP SUMMARY:
==250288==     in use at exit: 65,536 bytes in 1 blocks
==250288==   total heap usage: 60 allocs, 59 frees, 78,012 bytes allocated
==250288== 
==250288== LEAK SUMMARY:
==250288==    definitely lost: 65,536 bytes in 1 blocks
==250288==    indirectly lost: 0 bytes in 0 blocks
==250288==      possibly lost: 0 bytes in 0 blocks
==250288==    still reachable: 0 bytes in 0 blocks
==250288==         suppressed: 0 bytes in 0 blocks
==250288== Rerun with --leak-check=full to see details of leaked memory
==250288== 
==250288== ERROR SUMMARY: 1 errors from 1 contexts (suppressed: 0 from 0)
==250288== 
==250288== 1 errors in context 1 of 1:
==250288== Invalid read of size 1
==250288==    at 0x484ED16: strlen (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==250288==    by 0x48ECD30: __vfprintf_internal (vfprintf-internal.c:1517)
==250288==    by 0x48F7988: __vsprintf_internal (iovsprintf.c:95)
==250288==    by 0x10ACED: vsprintf (stdio2.h:52)
==250288==    by 0x10ACED: infof (main.c:91)
==250288==    by 0x10A966: main (main.c:121)
==250288==  Address 0x4ab5f90 is 0 bytes after a block of size 0 alloc'd
==250288==    at 0x4848899: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==250288==    by 0x10CF4E: readline (main.c:33)
==250288==    by 0x10A946: main (main.c:115)
==250288== 
==250288== ERROR SUMMARY: 1 errors from 1 contexts (suppressed: 0 from 0)
```

### Severity

high

### Version or commit hash

1647f21

### Environment

➜  ScorpionC2 git:(main) ✗ uname -orsn && gcc -v && valgrind --version  Linux pop-os 6.12.10-76061203-generic GNU/Linux Using built-in specs. COLLECT_GCC=gcc COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/11/lto-wrapper OFFLOAD_TARGET_NAMES=nvptx-none:amdgcn-amdhsa OFFLOAD_TARGET_DEFAULT=1 Target: x86_64-linux-gnu Configured with: ../src/configure -v --with-pkgversion='Ubuntu 11.4.0-1ubuntu1~22.04.2' --with-bugurl=file:///usr/share/doc/gcc-11/README.Bugs --enable-languages=c,ada,c++,go,brig,d,fortran,objc,obj-c++,m2 --prefix=/usr --with-gcc-major-version-only --program-suffix=-11 --program-prefix=x86_64-linux-gnu- --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --enable-bootstrap --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-vtable-verify --enable-plugin --enable-default-pie --with-system-zlib --enable-libphobos-checking=release --with-target-system-zlib=auto --enable-objc-gc=auto --enable-multiarch --disable-werror --enable-cet --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --enable-multilib --with-tune=generic --enable-offload-targets=nvptx-none=/build/gcc-11-2Y5pKs/gcc-11-11.4.0/debian/tmp-nvptx/usr,amdgcn-amdhsa=/build/gcc-11-2Y5pKs/gcc-11-11.4.0/debian/tmp-gcn/usr --without-cuda-driver --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu --with-build-config=bootstrap-lto-lean --enable-link-serialization=2 Thread model: posix Supported LTO compression algorithms: zlib zstd gcc version 11.4.0 (Ubuntu 11.4.0-1ubuntu1~22.04.2)  valgrind-3.18.1

### Confirmation

- [x] I have verified that this issue has not already been reported
- [x] I can reproduce this issue consistently

--- a/Makefile
+++ b/Makefile
@@ -6,7 +6,7 @@
 #
 
 .PHONY: build clean help rebuild test
-.SILENT: build help rebuild test
+.SILENT: build help rebuild test test-valigrind test-debug
 
 I_FLAGS 		:=  -Isrc-server/domain/encoders/services/xor \
 					-Isrc-server/domain/encoders/types \
@@ -18,6 +18,7 @@ I_FLAGS 		:=  -Isrc-server/domain/encoders/services/xor \
 					-Isrc-server/app/cli/input/readMode \
 					-Isrc-server/app/cli/input/safeget \
 					-Isrc-server/infra/fs \
+					-Isrc-server/app/cli/input \
 			
 OPTIMIZE_FLAGS 	:= 	-Os
 DEBUG_FLAGS 	:= 	-Og -dA -dD -ggdb -Wall -Wextra -Wformat=2 -Wshadow -Wundef
@@ -48,6 +49,7 @@ SRC_ENTRYPOINT	:=	src-server/app/main.c \
 					src-server/app/cli/ui/box/main.c \
 					src-server/app/cli/input/readMode/main.c \
 					src-server/app/cli/input/safeget/main.c \
+					src-server/app/cli/input/main.c \
 					src-server/infra/fs/main.c \
 
 all: help

--- a/src-server/app/cli/input/main.c
+++ b/src-server/app/cli/input/main.c
@@ -13,7 +13,6 @@
 void readline(InputSettings conf, string_t *out) {
     string_t prompt = conf.prompt;
     string_t histPath = conf.histPath;
-    int histLimit = conf.histLimit;
     
     printf("%s", prompt.s);
     fflush(stdout);
@@ -24,14 +23,22 @@ void readline(InputSettings conf, string_t *out) {
         
     };
     
-    if (Files.appendFile(histPath, &inputRaw) != 0) {
+    uchar_t _nl = '\n';
+    bytes_t newLine = {
+        .len = 1,
+        .b = &_nl
+        
+    };
+    
+    if ((Files.appendFile(histPath, &inputRaw) != 0) || (Files.appendFile(histPath, &newLine))) {
         Logger.warnln("Can't write last user input to history file");
         
     }
     
     out->len = input.len;
-    out->s = malloc(out->len);
+    out->s = malloc(out->len + 1);
     memcpy(out->s, input.s, out->len);
+    out->s[out->len] = '\0';
     
     free(input.s);
     return;

--- a/src-server/app/cli/input/safeget/main.c
+++ b/src-server/app/cli/input/safeget/main.c
@@ -8,11 +8,7 @@
 //  https://en.wikipedia.org/wiki/ANSI_escape_code
 //  https://stackoverflow.com/questions/7469139/what-is-the-equivalent-to-getch-getche-in-linux
 //  https://www.ibm.com/docs/pt-br/i/7.6.0?topic=functions-fgets-read-string
-// 
-//  TODO List:
-//      DONE  - Implement custom getchar without ECHO and CANONICAL modes
-//      INDEV - Implement custom read nor fgets above custom getchar
-//      TODO  - Adapt custom read/fgets to snread
+//
 
 #include "main.h"
 #include "../readMode/main.h"
@@ -24,45 +20,45 @@
 #include <unistd.h>
 #include <string.h>
 
-// ucgetchar - Ultra-incredible-handmade galactic empire engineer's Custom GETCHAR
-char ucgetchar(FILE *stream) {
+// ucfgets - Ultra-incredible-handmade galactic empire engineer's Custom FGETS
+char* ucfgets(string_t histPath, int promptSize) {
     TerminalModeToggler.turnCanonicalMode(FALSE);
     TerminalModeToggler.turnEchoMode(FALSE);
     
-    char ch = getchar();
-    
-    return ch;
-    
-    TerminalModeToggler.turnCanonicalMode(TRUE);
-    TerminalModeToggler.turnEchoMode(TRUE);
-    
-};
-
-// ucfgets - Ultra-incredible-handmade galactic empire engineer's Custom FGETS
-char* ucfgets(char *dest, string_t histPath, int promptSize) {
     size_t bufCap = 65536;
     size_t bufLen = 0;
     char *buf = malloc(bufCap);
     if (buf == NULL) {
-        return NULL;
+        goto cleanup_error;
         
     }
     
-    char ch;
-    long _n;
+    char ch = '\0';
     
     int currentHistLine;
     int histFileLines = Files.countLines(histPath);
     if (histFileLines < 0) {
-        return NULL;
+        goto cleanup_error;
         
     }
     
     currentHistLine = histFileLines;
 
-    for (int i = 0; (ch = ucgetchar(stdin)) != '\n'; i++) {
-        if ((bufLen + 1 + _n) > bufCap) {
-            while ((bufLen + 1 + _n) > bufCap) {
+    while (TRUE) {
+        ssize_t r = read(STDIN_FILENO, &ch, 1);
+        if (r <= 0) {
+            goto cleanup_error;
+            
+        }
+        
+        if (ch == 0xa) {
+            putchar(0xa);
+            goto cleanup;
+            
+        }
+        
+        if ((bufLen + 1) >= bufCap) {
+            while ((bufLen + 1) >= bufCap) {
                 bufCap <<= 1;
                 
             }
@@ -70,7 +66,7 @@ char* ucfgets(char *dest, string_t histPath, int promptSize) {
             char *newBuf = realloc(buf, bufCap);
             if (newBuf == NULL) {
                 free(buf);
-                return NULL;
+                goto cleanup_error;
                 
             }
             
@@ -79,18 +75,27 @@ char* ucfgets(char *dest, string_t histPath, int promptSize) {
         }
         
         if (ch == 0x1b) {
-            char ch2 = ucgetchar(stdin);
-            i++;
+            char ch2 = '\0';
+            ssize_t r2 = read(STDIN_FILENO, &ch2, 1);
+            if (r2 <= 0) {
+                goto cleanup_error;
+                
+            }
             
             if (ch2 == 0x5b) {
-                char ch3 = ucgetchar(stdin);
-                i++;
+                char ch3 = '\0';
+                ssize_t r3 = read(STDIN_FILENO, &ch3, 1);
+                if (r3 <= 0) {
+                    goto cleanup_error;
+                    
+                }
                 
                 bytes_t l;
-                char *histLine;
+                char *histLine = NULL;
                 switch (ch3) {
                     case 0x41: // Arrow up
-                        if (Files.getLine(histPath, currentHistLine + 1, &l) != 0) {
+                        if (histFileLines == 0) continue;
+                        if (Files.getLine(histPath, (currentHistLine - 1 + histFileLines) % histFileLines, &l) != 0) {
                             Logger.warnln("Can't parse hist line");
                             continue;
                             
@@ -105,17 +110,30 @@ char* ucfgets(char *dest, string_t histPath, int promptSize) {
 
                         memcpy(histLine, l.b, l.len);
                         histLine[strcspn(histLine, "\n")] = '\0';
-                        
-                        printf("\x1b[1G\x1b[%dC%s", promptSize, histLine);
-                        
-                        free(histLine);
                         free(l.b);
+                        
+                        bufLen = strlen(histLine);
+                        if (bufLen + 1 >= bufCap) {
+                            while (bufLen + 1 >= bufCap) {
+                                bufCap <<= 1;
+                            }
+                            buf = realloc(buf, bufCap);
+                        }
+                        
+                        memcpy(buf, histLine, bufLen);
+                        buf[bufLen] = '\0';
+                        free(histLine);
+                        
+                        printf("\x1b[1G\x1b[%dC%s\x1b[K", promptSize, buf);
+                        fflush(stdout);
                         currentHistLine--;
                         
                         continue;
                     
                     case 0x42:
-                        if (Files.getLine(histPath, currentHistLine - 1, &l) != 0) {
+                        if (histFileLines == 0) continue;
+
+                        if (Files.getLine(histPath, (currentHistLine + 1) % histFileLines, &l) != 0) {
                             Logger.warnln("Can't parse hist line");
                             continue;
                             
@@ -130,12 +148,25 @@ char* ucfgets(char *dest, string_t histPath, int promptSize) {
                         
                         memcpy(histLine, l.b, l.len);
                         histLine[strcspn(histLine, "\n")] = '\0';
-                        
-                        printf("\x1b[1G\x1b[%dC%s", promptSize, histLine);
-                        
-                        free(histLine);
                         free(l.b);
-                        currentHistLine--;
+                        
+                        
+                        bufLen = strlen(histLine);
+                        if (bufLen + 1 >= bufCap) {
+                            while (bufLen + 1 >= bufCap) {
+                                bufCap <<= 1;
+                            }
+                            buf = realloc(buf, bufCap);
+                        }
+                        
+                        memcpy(buf, histLine, bufLen);
+                        buf[bufLen] = '\0';
+                        free(histLine);
+                        
+                        printf("\x1b[1G\x1b[%dC%s\x1b[K", promptSize, buf);
+                        fflush(stdout);
+                        
+                        currentHistLine++;
                         
                         continue;
                         
@@ -144,45 +175,67 @@ char* ucfgets(char *dest, string_t histPath, int promptSize) {
 
                 }
             }
+        } else if (ch == 0x7f || ch == 0x08) {
+            if (bufLen > 0) {
+                bufLen--;
+                buf[bufLen] = '\0';
+            
+            }
+            
+            printf("\x1b[1G\x1b[%dC%s\x1b[K", promptSize, buf);
+            fflush(stdout);
+            continue;
+            
         } else {
-            buf[i] = ch;
+            buf[bufLen++] = ch;
+            buf[bufLen] = '\0';
+            putc(ch, stdout);
+            fflush(stdout);
             
         }
         
-        _n++;
-        bufLen += _n;
-        
     }
     
-    dest = malloc(bufLen);
-    memcpy(dest, buf, bufLen);
+    buf[bufLen] = '\0';
     
-    free(buf);
-    return dest;
+    printf("\x1b[1G\x1b[%dC%s\x1b[K", promptSize, buf);
+    fflush(stdout);
+    
+    cleanup:
+        TerminalModeToggler.turnCanonicalMode(TRUE);
+        TerminalModeToggler.turnEchoMode(TRUE);
+        return buf;
+    
+    cleanup_error:
+        TerminalModeToggler.turnCanonicalMode(TRUE);
+        TerminalModeToggler.turnEchoMode(TRUE);
+        free(buf);
+        return NULL;
     
 };
 
 string_t ucread(string_t histPath, int promptSize) {
     string_t out;
-    char *buf;
+    char *buf = ucfgets(histPath, promptSize);
     
-    if (ucfgets(buf, histPath, promptSize) == NULL) {
+    if (buf == NULL) {
         out.len = 0;
         out.s = NULL;
         return out;
         
     }
     
-    buf[strcspn(buf, "\n")] = '\0';
     out.len = strlen(buf);
-    out.s = malloc(out.len);
+    out.s = malloc(out.len + 1);
     if (out.s == NULL) {
         out.len = 0;
+        free(buf);
         return out;
         
     }
     
     memcpy(out.s, buf, out.len);
+    out.s[out.len] = '\0';
     free(buf);
     
     return out;

--- a/src-server/app/cli/input/safeget/main.h
+++ b/src-server/app/cli/input/safeget/main.h
@@ -10,11 +10,8 @@
 #include <stdio.h>
 
 typedef struct {    
-    // ucgetchar - Ultra-incredible-handmade galactic empire engineer's Custom GETCHAR
-    char (*ucgetchar)(FILE *stream);
-    
     // ucfgets - Ultra-incredible-handmade galactic empire engineer's Custom FGETS
-    char* (*ucfgets)(char *dest, string_t histPath, int promptSiz);
+    char* (*ucfgets)(string_t histPath, int promptSiz);
     
     /*
      * Return rules:

--- a/src-server/infra/fs/main.c
+++ b/src-server/infra/fs/main.c
@@ -25,9 +25,9 @@ int appendFile(string_t path, bytes_t *src) {
 };
 
 int countLines(string_t path) {
-    int count;
-    int c;
-    if (!path.s || !path.len) {
+    int count = 0;
+    int c = 0;
+    if (path.s == NULL || path.len <= 0) {
         return -1;
         
     }
@@ -52,7 +52,7 @@ int countLines(string_t path) {
 }
 
 int getLine(string_t path, int line, bytes_t *out) {
-    if (!path.s || !path.len) return 1;
+    if (path.s == NULL || path.len == 0) return 1;
     int lines = countLines(path);
     if (lines < 0) {
         return 1;
@@ -96,6 +96,6 @@ int getLine(string_t path, int line, bytes_t *out) {
 FsInstance Files = {
     .countLines = countLines,
     .appendFile = appendFile,
-    // .getLine = 
+    .getLine = getLine
     
 };

